query-engine/connectors/sql-query-connector/src/database/operations/read.rs

```diff
@@ -54,6 +56,81 @@ pub(crate) async fn get_single_record(
 }
 
 pub(crate) async fn get_many_records(
+    conn: &dyn Queryable,
+    model: &Model,
+    query_arguments: QueryArguments,
+    selected_fields: &FieldSelection,
+    aggr_selections: &[RelAggregationSelection],
+    relation_load_strategy: RelationLoadStrategy,
+    ctx: &Context<'_>,
+) -> crate::Result<ManyRecords> {
+    match relation_load_strategy {
+        RelationLoadStrategy::Join => {
+            get_many_records_joins(conn, model, query_arguments, selected_fields, aggr_selections, ctx).await
+        }
+        RelationLoadStrategy::Query => {
+            get_many_records_wo_joins(
+                conn,
+                model,
+                query_arguments,
+                &ModelProjection::from(selected_fields),
+                aggr_selections,
+                ctx,
+            )
+            .await
+        }
+    }
+}
+
+pub(crate) async fn get_many_records_joins(
+    conn: &dyn Queryable,
+    _model: &Model,
+    query_arguments: QueryArguments,
+    selected_fields: &FieldSelection,
+    _aggr_selections: &[RelAggregationSelection],
+    ctx: &Context<'_>,
+) -> crate::Result<ManyRecords> {
+    let field_names: Vec<_> = selected_fields.db_names().collect();
+    let idents = selected_fields.type_identifiers_with_arities();
+    let meta = column_metadata::create(field_names.as_slice(), idents.as_slice());
+
+    let rs_indexes = get_relation_selection_indexes(selected_fields.relations().collect(), &field_names);
+    let mut records = ManyRecords::new(field_names.clone());
+
+    if let Some(0) = query_arguments.take {
+        return Ok(records);
+    };
+
+    match ctx.max_bind_values {
+        Some(chunk_size) if query_arguments.should_batch(chunk_size) => {
+            return Err(SqlError::QueryParameterLimitExceeded(
+                "Joined queries cannot be split into multiple queries just yet. If you encounter this error, please open an issue"
```

Comment:
We discussed how this could trigger by a user exceeding the number of bind params artificially (f.i. by dynamically composing wrong a query) and we agreed this wouldn't be subject of an issue on our end. As such this message can be misleading and frame as if we did something wrong with the implementation. A more appropriate tactic might be ignoring batching altogether and letting the code reach the DB, bubbling up to the user any potential query parameter limit exceeded.